C-Plus-Plus/ds/CircularQueue.cpp

```diff
@@ -0,0 +1,144 @@
+        for(int i = 0; i < size; i++) {
+        	arr[i] = 0;
+		}
+    }
+    //check for empty queue
+    bool isEmpty() {
+        if (front == -1 && rear == -1) 
+            return true;
+        else 
+            return false;
+    }
+    //check for overflow in queue
+    bool isFull() {
+        if (((rear + 1) % size) == front) 
+            return true;
+        else
+            return false;
+    }
+	void enqueue(int);
+	int dequeue();
+	void display();
+};
+
+//inserting values in queue
+void CircularQ:: enqueue(int value) {
+	if (isFull()) {
+        cout << "\n!!!Overflow State!!!\n";
+        return;
+    } else if (isEmpty()) {
+        rear = 0;
+        front = 0;
+        arr[rear] = value;
+    } else {
+        rear = (rear + 1) % size;
+        arr[rear] = value;
+    }		
+}
+
+//deleting values from queue
+int CircularQ:: dequeue() {
+	int element = 0;
+        if(isEmpty()) {
+        	cout << "\n!!!Empty Queue!!!\n";
+        	return element;
+		} else if(front == rear) {
+			element = arr[front];
+			front = rear = - 1;
+			return element;
+		}  else {
+			cout << "\nFront value: " << front << endl;
+			element = arr[front];
+			arr[front] = 0;
+			front = (front + 1) % size;
+			return element;
+		}
+}
+
+//displaying contents of queue
+void CircularQ::display() {
+	for (int i = 0; i < size; i++){
+    	cout << arr[i] << " ";
+	}
+}
+
+int main() {
+	int QueueSize;
+	cout << "Enter size of Circular Queue\n";
+	cin >> QueueSize;
+    CircularQ cq(QueueSize);
+    int choice, value;
+    
+    do {
+		cout << "\n\nSelect from the following options:\n";
+    	cout << "1. Enqueue\n"
+            "2. Dequeue\n"
+            "3. isEmpty\n"
+            "4. isFull\n"
+            "5. display\n"
+            "Press 0 to exit.\n";
+    	cin >> choice;
+        switch (choice) {
+            case 0:
+                break;
+            
+            case 1:
+                cout << "\nInput value to Enqueue\n";
+                cin >> value;
+                cq.enqueue(value);
+                break;
+            
+            case 2: 
+                cout << "\nDequeue Operation...\n";
+                cout << cq.dequeue() << endl;
+                break;
+            
+            case 3:
+                if (cq.isEmpty())
+                    cout << "\nQueue is Empty\n";
+                else 
+                    cout << "\nQueue is not Empty\n";
+                break;
+            
+            case 4:
+                if (cq.isFull())
+                    cout << "\nOverflow State\n";
+                else 
+                    cout << "\nQueue is not Full\n";
+                break;
+            
+            case 5: 
+                cout << "\nQueue elements displayed: \n";
+                cq.display();
+                break;
+            
+            default:
+                cout << "\n!!!Invalid choice. Try Again!!!\n";
+                break;
+        }
+    } while (choice != 0);
+
+    return 0;
+}
```

Comment:
Add sample Input/Output demonstrating all the functionalities of a circular queue.

Reply:
Do you mean I should change it to one which has default input values rather than a dynamic one? And how should I add a sample output?

Reply:
No, inputs are fine. Just run the code for a sample test case and then add the input and output in commented form at the end of your program. You may refer to [this](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/ds/Vector.cpp).

Reply:
Okay. Doing it.